SD_Card_Driver/Core/Inc/LOS_D_storage_device.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef LOS_D_STORAGE_DEVICE_HPP_
+#define LOS_D_STORAGE_DEVICE_HPP_
+
+class StorageDevice {
+	public:
+		/* Initialize the storage device */
+		virtual int init() = 0;
+
+		/* Check if a location exists in the storage device */
+		virtual bool checkExist(const char* location) = 0;
+
+		/* Read bytes from a location with an optional offset */
+		virtual int read(const char* location, uint8_t* data, size_t size, size_t offset = 0) = 0;
+
+		/* Write bytes from a location with an optional offset */
+		virtual int write(const char* location, uint8_t* data, size_t size, size_t offset = 0) = 0;

```

Comment:
Or even better, can we use a const reference here?

Maybe we should also be looking into the use of std::string or std::span? It would depend on how much inclusion of the std library blows up the code or what the downsides are. I'm not sure if it is worth the benefits they provide

Reply:
Yea `char[]` would be a better type since FatFS doesn't really care what type of pointer we pass in. By const reference do you mean `const char[] &data`?

Reply:
Yeah, that's what I was referring to. I cant remember if that's legal though